### PR TITLE
Deprecate using shortcut notation for user_model in favor of FQCN

### PR DESCRIPTION
**Pull request by franmomu**
## Subject

Ref: https://github.com/sonata-project/SonataAdminBundle/pull/6407#discussion_r494063621

If we use FQCN notation for the `user_model` option in `GenerateObjectAclCommand`, there is no need for the `$registry` dependency.

I am targeting this branch, because these changes are BC.

## Changelog

```markdown
### Deprecated
- Deprecate using shortcut notation for user_model in favor of FQCN in `GenerateObjectAclCommand`
```

## To do

- [x] Add an upgrade note.

### CODE REVIEW

File: src/Command/GenerateObjectAclCommand.php [hunk 1]
```diff
@@ -94,7 +97,7 @@ public function configure()
             ->addOption('object_owner', null, InputOption::VALUE_OPTIONAL, 'If set, the task will set the object owner for each admin.')
             // NEXT_MAJOR: Remove "user_entity" option.
             ->addOption('user_entity', null, InputOption::VALUE_OPTIONAL, '<error>DEPRECATED</error> Use <comment>user_model</comment> option instead.')
-            ->addOption('user_model', null, InputOption::VALUE_OPTIONAL, 'Shortcut notation like <comment>AcmeDemoBundle:User</comment>. If not set, it will be asked the first time an object owner is set.')
+            ->addOption('user_model', null, InputOption::VALUE_OPTIONAL, 'Fully qualified class name <comment>App\Entity\User</comment>. If not set, it will be asked the first time an object owner is set.')
```
Comment (phansys) [inline src/Command/GenerateObjectAclCommand.php:100]: IMO, we should avoid terms ("Entity", "Document") that are tied to a specific model manager.

Comment (VincentLanglet) [reply src/Command/GenerateObjectAclCommand.php:100]: Model then ?

Comment (phansys) [reply src/Command/GenerateObjectAclCommand.php:100]: I think yes. We haven't many choices.

File: src/Command/GenerateObjectAclCommand.php [hunk 2]
```diff
@@ -226,25 +229,44 @@ protected function getUserEntityClass(InputInterface $input, OutputInterface $ou
...
+
+//                throw new \InvalidArgumentException(sprintf(
+//                    'The "user_model" name be a fully qualified class name'
+//                    .' ("%s" given, expecting something like App\Model\User)',
+//                    $userModelFromInput
+//                ));
+
+                [$userBundle, $userModel] = Validators::validateEntityName($userModelFromInput);
+
+                // Entity exists?
```
Comment (phansys) [inline src/Command/GenerateObjectAclCommand.php:259]:
```suggestion
                // Model exists?
```

**Review by phansys — CHANGES_REQUESTED** [review]

**Review by phansys — APPROVED** [review]

File: src/Command/GenerateObjectAclCommand.php [hunk 3]
```diff
@@ -58,6 +59,8 @@ class GenerateObjectAclCommand extends QuestionableCommand
     private $registry;
 
     /**
+     * NEXT_MAJOR: Remove $registry argument.
```
Comment (jordisala1991) [inline src/Command/GenerateObjectAclCommand.php:62]: We should remove the dependency from the DI php file

**Review by jordisala1991 — CHANGES_REQUESTED** [review]

**Review by phansys — APPROVED** [review]

**Review by jordisala1991 — APPROVED** [review]

**Comment by phansys** [comment]
Thank you @franmomu!

**Pull request merged**